app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldActivityFeedViewModel.kt

```diff
@@ -194,8 +242,14 @@ class DeviceShieldActivityFeedViewModel @Inject constructor(
                 }
 
                 if (firstInBucket && showHeadings) {
-                    sourceData.add(TrackerFeedItem.TrackerFeedItemHeader(item.trackerCompanySignal.tracker.timestamp))
-                        .also { firstInBucket = false }
+                    sourceData.add(
+                        TrackerFeedItem.TrackerFeedItemHeader(
+                            timeDiffFormatter.formatTimePassedInDays(
```

Comment:
Moved the formatting here so we can have the same content for `Today` / `Yesterday` / etc to avoid unnecessary updates in the RecyclerView.